### DISCUSSION TITLE
Extending chakra UI to make reusable "blocks" packages

### DISCUSSION THREAD

**Discussion by wiredmatt** [chakra-ui/chakra-ui#6026] [Q&A] [answered]
Hi, I've been using chakra for a few years in many projects. But I find myself copy-pasting the majority of the `blocks` I built (A React component that uses a lot of Chakra UI components in it). So I wanted to create a package I can just add using `yarn install my-blocks-package` containing these UI blocks that I have built throughout the years.

I've tried a lot of things but can't seem to get it to work, I think it might be due to compiler options, but surely you guys will be able to propose something that might make it work.

[This](https://github.com/system32uwu/turborepo-chakra-ui) is the current repository where I'm trying to make it work, but I'm getting an error from webpack `react__WEBPACK_IMPORTED_MODULE_0__.startTransition is not a function`.

The repo uses turbo workspaces, in order to run the Next App there just run `yarn dev`.

Thanks in advance folks, cheers.

**Comment by leplay** [comment, marked as answer]
I think it's caused by Chakra V2, it is not backwards compatible with React v17 and lower.  You can try to upgrade React or downgrade Chakra UI to 1.x.
https://chakra-ui.com/guides/migration#1-update-your-dependencies

**Reply by wiredmatt** [reply]
Thank you, it was indeed the react version.

**Comment by 6qat** [comment]
You can solve this by adding "use client" in the beginning of your file.
